src/vs/workbench/browser/parts/quickopen/quickopen.ts

```diff
@@ -22,15 +22,15 @@ export class GlobalQuickOpenAction extends Action {
 	public static ID = 'workbench.action.quickOpen';
 	public static LABEL = nls.localize('quickOpen', "Go to File...");
 
-	constructor(id: string, label: string, @IQuickOpenService private quickOpenService: IQuickOpenService) {
+	constructor(id: string, label: string, @IQuickOpenService private quickOpenService: IQuickOpenService, @ICommandService private commandService: ICommandService) {
 		super(id, label);
 
 		this.order = 100; // Allow other actions to position before or after
 		this.class = 'quickopen';
 	}
 
 	public run(): TPromise<any> {
-		this.quickOpenService.show(null);
+		this.commandService.executeCommand('vscode.quickOpen');
```

Comment:
I don't really think this is madness per se, but okay. I see what you mean. It's not okay to have both a command and an action that do the same thing. But for my piece of mind and since I've just started working on VS Code, can you please detail what;s the difference between an API command, a command and an action? what are they each used for?